File: villa_tiktok_slides/create_slides.py
#!/usr/bin/env python3
"""Create TikTok-ready slideshow images for Villa Lithos"""

from PIL import Image, ImageDraw, ImageFont
import os

PHOTOS_DIR = os.path.join(os.path.dirname(__file__), 'photos')
OUTPUT_DIR = os.path.join(os.path.dirname(__file__), 'final')
os.makedirs(OUTPUT_DIR, exist_ok=True)

# TikTok dimensions (9:16)
WIDTH = 1080
HEIGHT = 1920

# Slide definitions: (photo_file, text_lines, text_position)
SLIDES = [
    {
        'photo': 'exterior.jpg',
        'lines': [
            'My friends think',
            "I'm rich...",
            '',
            'But this luxury villa',
            'costs LESS per person',
            'than most hotels 🏛️'
        ],
        'position': 'top'
    },
    {
        'photo': 'pool.jpg',
        'lines': [
            '22 guests',
            '9 bedrooms',
            'Private infinity pool',
            '15 min from Athens airport ✈️'
        ],
        'position': 'top'
    },
    {
        'photo': 'aerial.jpg',
        'lines': [
            'Private padel court 🎾',
            'Finnish barrel sauna 🧖',
            'Heated infinity pool 🏊',
            'Jacuzzi • Gym • Spa',
            '5,000m² private estate'
        ],
        'position': 'top'
    },
    {
        'photo': 'bathtub.jpg',
        'lines': [
            '€8,000 per week',
            '÷ 22 people',
            '= €52 per person/night',
            '',
            'That\'s LESS than a hostel 🤯'
        ],
        'position': 'center'
    },
    {
        'photo': 'fireplace.jpg',
        'lines': [
            'Hotels in Mykonos:',
            '€300+ per night per room',
            '',
            'Villa Lithos:',
            '€52/night with YOUR',
            'own infinity pool 💎'
        ],
        'position': 'top'
    },
    {
        'photo': 'hero.webp',
        'lines': [
            'Villa Lithos',
            'Porto Rafti, Greece',
            '',
            '🔗 Link in bio',
            '📩 DM for availability'
        ],
        'position': 'center'
    }
]


def load_font(size):
    """Try to load a bold font, fall back to default"""
    font_paths = [
        '/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf',
        '/usr/share/fonts/truetype/liberation/LiberationSans-Bold.ttf',
        '/usr/share/fonts/truetype/freefont/FreeSansBold.ttf',
    ]
    for fp in font_paths:
        if os.path.exists(fp):
            return ImageFont.truetype(fp, size)
    return ImageFont.load_default()


def create_slide(slide_def, index):
    """Create a single TikTok slide"""
    photo_path = os.path.join(PHOTOS_DIR, slide_def['photo'])
    
    if not os.path.exists(photo_path):
        print(f"  ⚠️  Photo not found: {photo_path}")
        return None
    
    # Open and resize photo to fill 9:16
    img = Image.open(photo_path)
    
    # Calculate crop to fill 9:16
    target_ratio = WIDTH / HEIGHT  # 0.5625
    img_ratio = img.width / img.height
    
    if img_ratio > target_ratio:
        # Image is wider, crop sides
        new_width = int(img.height * target_ratio)
        left = (img.width - new_width) // 2
        img = img.crop((left, 0, left + new_width, img.height))
    else:
        # Image is taller, crop top/bottom
        new_height = int(img.width / target_ratio)
        top = (img.height - new_height) // 2
        img = img.crop((0, top, img.width, top + new_height))
    
    img = img.resize((WIDTH, HEIGHT), Image.LANCZOS)
    
    # Add dark overlay for text readability
    overlay = Image.new('RGBA', (WIDTH, HEIGHT), (0, 0, 0, 0))
    draw = ImageDraw.Draw(overlay)
    
    if slide_def['position'] == 'top':
        # Gradient from top
        for y in range(0, HEIGHT // 2):
            alpha = int(180 * (1 - y / (HEIGHT // 2)))
            draw.rectangle([(0, y), (WIDTH, y + 1)], fill=(0, 0, 0, alpha))
    elif slide_def['position'] == 'center':
        # Full semi-transparent overlay
        draw.rectangle([(0, 0), (WIDTH, HEIGHT)], fill=(0, 0, 0, 100))
    
    # Composite overlay
    img = img.convert('RGBA')
    img = Image.alpha_composite(img, overlay)
    
    # Add text
    draw = ImageDraw.Draw(img)
    font_large = load_font(56)
    font_small = load_font(44)
    
    lines = slide_def['lines']
    line_height = 72
    total_height = len(lines) * line_height
    
    if slide_def['position'] == 'top':
        y_start = 200
    elif slide_def['position'] == 'center':
        y_start = (HEIGHT - total_height) // 2
    else:
        y_start = HEIGHT - total_height - 200
    
    for i, line in enumerate(lines):
        if not line:
            continue
        
        font = font_large if i == 0 or (index == 0 and i < 2) else font_small
        
        # Get text size
        bbox = draw.textbbox((0, 0), line, font=font)
        text_width = bbox[2] - bbox[0]
        x = (WIDTH - text_width) // 2
        y = y_start + i * line_height
        
        # Draw text shadow
        draw.text((x + 3, y + 3), line, fill=(0, 0, 0, 200), font=font)
        # Draw main text
        draw.text((x, y), line, fill=(255, 255, 255, 255), font=font)
    
    # Convert back to RGB for saving
    img = img.convert('RGB')
    
    output_path = os.path.join(OUTPUT_DIR, f'slide-{index + 1}.jpg')
    img.save(output_path, 'JPEG', quality=95)
    print(f"  ✅ Slide {index + 1}: {output_path}")
    return output_path


def main():
    print("🎬 Creating Villa Lithos TikTok Slides\n")
    
    slides = []
    for i, slide_def in enumerate(SLIDES):
        result = create_slide(slide_def, i)
        if result:
            slides.append(result)
    
    print(f"\n📱 Created {len(slides)}/{len(SLIDES)} slides")
    print(f"📁 Output: {OUTPUT_DIR}/")
    
    if slides:
        print("\n📋 Caption for TikTok:")
        caption = """My friends think I'm rich but this luxury villa in Greece costs less per person than most hotels 🏛️✨

22 guests • 9 bedrooms • infinity pool • padel court • sauna
Only €52/person per night!

📍 Villa Lithos, Porto Rafti - 15 min from Athens Airport

#luxuryvilla #greecevilla #travelgreece #villagoals #airbnbfinds #grouptravel #europetravel #luxurytravel #splitcost #budgetluxury #travelhack #smarttravel #squadgoals #friendtrip #portorafti #athensgreece #greeksummer #visitgreece"""
        print(caption)
        
        # Save caption
        with open(os.path.join(OUTPUT_DIR, 'caption.txt'), 'w') as f:
            f.write(caption)


if __name__ == '__main__':
    main()

File: villa-lithos-tiktok/create-slides.js
#!/usr/bin/env node
const fs = require('fs');
const path = require('path');
const { createCanvas, loadImage } = require('canvas');

const SLIDE_W = 1080;
const SLIDE_H = 1920;

const slides = [
  {
    img: 'slides/slide-1.png', // hero - villa exterior
    text: "My friends think I'm rich,\nbut this luxury villa in Greece\ncosts less per person\nthan most hotels"
  },
  {
    img: 'slides/slide-2.png', // pool
    text: "22 people, 9 bedrooms,\nprivate infinity pool,\n15 min from Athens airport"
  },
  {
    img: 'slides/slide-3.png', // wellness/spa
    text: "Private paddle court, sauna,\n5000m² of pure luxury"
  },
  {
    img: 'slides/slide-4.png', // exterior
    text: "€8000 per week\n÷ 22 people\n= €360 per person\nfor THE WHOLE WEEK"
  },
  {
    img: 'slides/slide-5.png', // living/dining
    text: "Meanwhile hotels in Mykonos\ncharge €300+ per NIGHT\nper room"
  },
  {
    img: 'slides/slide-6.png', // exterior pool
    text: "Villa Lithos\nPorto Rafti, Greece\nlink in bio"
  }
];

async function createSlide(slideInfo, index) {
  const canvas = createCanvas(SLIDE_W, SLIDE_H);
  const ctx = canvas.getContext('2d');

  // Load and crop-to-fill the image
  const img = await loadImage(slideInfo.img);
  
  // Calculate crop to fill 1080x1920 (center crop)
  const targetRatio = SLIDE_W / SLIDE_H;
  const imgRatio = img.width / img.height;
  
  let sx, sy, sw, sh;
  if (imgRatio > targetRatio) {
    // Image is wider - crop sides
    sh = img.height;
    sw = img.height * targetRatio;
    sx = (img.width - sw) / 2;
    sy = 0;
  } else {
    // Image is taller - crop top/bottom
    sw = img.width;
    sh = img.width / targetRatio;
    sx = 0;
    sy = (img.height - sh) / 2;
  }
  
  // Draw image filling canvas
  ctx.drawImage(img, sx, sy, sw, sh, 0, 0, SLIDE_W, SLIDE_H);
  
  // Add dark gradient overlay - heavier at top where text goes, lighter below to show villa
  const gradient = ctx.createLinearGradient(0, 0, 0, SLIDE_H);
  gradient.addColorStop(0, 'rgba(0,0,0,0.65)');
  gradient.addColorStop(0.30, 'rgba(0,0,0,0.35)');
  gradient.addColorStop(0.45, 'rgba(0,0,0,0.05)');
  gradient.addColorStop(0.75, 'rgba(0,0,0,0.0)');
  gradient.addColorStop(1, 'rgba(0,0,0,0.3)');
  ctx.fillStyle = gradient;
  ctx.fillRect(0, 0, SLIDE_W, SLIDE_H);

  // Text overlay
  const lines = slideInfo.text.split('\n');
  const wordCount = slideInfo.text.replace(/\n/g, ' ').split(/\s+/).length;
  
  let fontSize;
  if (wordCount <= 8)       fontSize = 72;
  else if (wordCount <= 15) fontSize = 62;
  else                      fontSize = 52;

  // Special: slide 6 (CTA) gets bigger text
  if (index === 5) fontSize = 78;
  // Slide 4 (math) gets medium
  if (index === 3) fontSize = 68;
  
  const lineHeight = fontSize * 1.35;
  const outlineWidth = Math.round(fontSize * 0.12);

  ctx.font = `bold ${fontSize}px "Arial"`;
  ctx.textAlign = 'center';
  ctx.textBaseline = 'middle';

  const totalH = lines.length * lineHeight;
  // Position text in top third - keeps villa photos visible below
  const startY = (SLIDE_H * 0.18) - (totalH / 2) + (totalH / 2);
  const x = SLIDE_W / 2;

  for (let i = 0; i < lines.length; i++) {
    const y = startY + (i * lineHeight);
    const line = lines[i].trim();

    // Shadow for depth
    ctx.shadowColor = 'rgba(0,0,0,0.8)';
    ctx.shadowBlur = 15;
    ctx.shadowOffsetX = 3;
    ctx.shadowOffsetY = 3;
    
    // Black outline
    ctx.strokeStyle = '#000000';
    ctx.lineWidth = outlineWidth;
    ctx.lineJoin = 'round';
    ctx.miterLimit = 2;
    ctx.strokeText(line, x, y);

    // White fill
    ctx.shadowBlur = 0;
    ctx.shadowOffsetX = 0;
    ctx.shadowOffsetY = 0;
    ctx.fillStyle = '#FFFFFF';
    ctx.fillText(line, x, y);
  }

  // Save
  const outPath = `slides/final-${index + 1}.png`;
  fs.writeFileSync(outPath, canvas.toBuffer('image/png'));
  console.log(`✅ Slide ${index + 1}: ${outPath} (${lines.length} lines, font ${fontSize}px)`);
}

async function main() {
  console.log('🎬 Creating Villa Lithos TikTok Slides...\n');
  
  for (let i = 0; i < slides.length; i++) {
    await createSlide(slides[i], i);
  }
  
  console.log('\n🎉 All 6 TikTok slides created!');
  console.log('📁 Output: slides/final-1.png through final-6.png');
  console.log('📐 Format: 1080x1920 (TikTok portrait)');
}

main().catch(err => { console.error(err); process.exit(1); });
